### DISCUSSION TITLE
Suggesting what to do for warning `you should not extend "group:" presets`

### DISCUSSION THREAD

**Discussion by jamesbraza** [renovatebot/renovate#31628] [Suggest an Idea]
### Tell us more.

https://github.com/renovatebot/renovate/pull/9836 added a warning `you should not extend "group:" presets`. This is a fine warning, but it would be good to suggest how to resolve this.

Like if one has a config like this (which will get a warning by `renovate-config-validator`):

```json5
{
  $schema: "https://docs.renovatebot.com/renovate-schema.json",
  extends: ["config:recommended"],
  packageRules: [
    {
      extends: ["group:allNonMajor"],
      automerge: true,
    },
  ],
}
```

It would be good to expand the warning's message to also suggest the proper resolution

**Comment by aaomidi** [comment]
Don't extend the rule basically, write it out yourself: https://docs.renovatebot.com/presets-group/#groupallnonmajor

**Reply by jamesbraza** [reply]
Thanks for the answer, it's too bad they don't let us extend

**Reply by aaomidi** [reply]
To be fair it is just a warning…

**Reply by jamesbraza** [reply]
Sadly the warning leads to a nonzero `pre-commit` error, which breaks my CI. So even though it's a warning in version `38.94.3`, for me it's basically an error

**Reply by aaomidi** [reply]
I would love to know what the justification of not extending these rules are. I think it’s something that should definitely be linked to in that error. I can maybe come up with a few reasons but most of them apply to the other presets too.

**Reply by 1oglop1** [reply]
@RahulGautamSingh is there any chance you could answer this?

**Reply by rarkins** [reply]
It's because there are many cases where it simply won't work. It wasn't intended behavior to be able to do it, and often causes failures, so we recommend you don't do it.
